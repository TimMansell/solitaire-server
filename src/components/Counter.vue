<template>
  <span>
    {{ formattedCount }}
  </span>
</template>

<script>
import numeral from 'numeral';

export default {
  name: 'Counter',
  props: {
    number: {
      type: Number,
      default: 0,
    },
  },
  computed: {
    formattedCount() {
      const { number } = this;

      return numeral(number).format('0,0');
    },
  },
};
</script>
