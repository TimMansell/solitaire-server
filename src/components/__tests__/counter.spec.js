import { shallowMount } from '@vue/test-utils';
import Counter from '@/components/Counter.vue';

describe('Counter.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Counter);

    expect(wrapper).toMatchSnapshot();
  });

  it('should correctly format 1000 as 1,000', () => {
    const wrapper = shallowMount(Counter, {
      propsData: {
        number: 1000,
      },
    });

    expect(wrapper.find('[data-test="counter"]').text()).toContain('1,000');
  });

  it('should correctly format 1000000 as 1,000,000', () => {
    const wrapper = shallowMount(Counter, {
      propsData: {
        number: 1000000,
      },
    });

    expect(wrapper.find('[data-test="counter"]').text()).toContain('1,000,000');
  });
});
