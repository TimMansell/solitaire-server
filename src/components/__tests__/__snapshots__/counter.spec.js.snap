// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue matches snapshot 1`] = `<span data-test="counter">0</span>`;
